Components/Modal/Modal.tsx

```diff
@@ -0,0 +1,195 @@
+Aeag81OdlduwsE5YH2QJObHbbDQDAeZg8zNdhluwwE4UH2IFOLnbbTwGIeYg8xNdltuw0s4gH2AFObHbfjwGQeXg8ve7Drdh2YxIPrwWUvdt7OA4N5aD207Mnu231gEA+
+sB5a9eQO8AUBnHlYPK7TmLfAWAN14UD2o8JY3wZsAXOQh9ZDCLd4GbwNwwXNb/1hFPH+01p4ufFeo4Km97MLqfYx4ni98V2ADP7fW/mzrH6tIx/9O4St/yXp//mqt/XLl
+owJ7ULK+HuUK3lOyvh7lCjhFyVKu4B4lS7kCHrRzyVKu4GM7lyzlCrhkx5KlXMFxO5Ys5QroYqeSpVzBeTuVLOUK6GqHkqVcweN2KFnKFTBE5ZKlXMF1lUuWcgUMVbFkK
+VfQT8WSpVwBU1QqWcoV9FepZClXwFQVSpZyBeNUKFnKFbBE5pKlXMF4mUuWcgUslbFkKVcwT8aSpVwBIWQqWcoVzJepZClXQCgZSpZyBetkKFnKFRBS5JKlXMF6kUuWcg
+WEFrFkKVcQR8SSpVwBKUQqWcoVxBOpZClXQCoRSpZyBXFFKFnKFZDSypKlXEF8K0uWcgWktqJkKVeQx4qSpVwBJcwsWcoV5DOzZClXQCkzSpZyBXnNKFnKFVDSyJKlXEF
++I0uWcgWUNqJkKVdQx4iSpVwBW+hZspQrqKdnyVKugK30KFnKFdTVo2QpV8CWrpQs5Qrqu1KylCtga4+ULOUK9vFIyVKuANq5kqVcwX7OlCzlCuCNIyVLuYJ9HSlZyhXA
+DfdKlnIF3CtZyhXAHbdKlnIFvLpVspQrgAPelizlCvje25KlXAGc8Km19txae1r8O4CYntrLG/Fp8e8AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAgOX+Bzv0VJug7DKZAAAAAElFTkSu
+QmCC) no-repeat center right;
+    height: 20px;
+    width: 20px;
+    background-size: 20px;
+    padding-left: 20px;
+    cursor: pointer;
+    margin-left: auto;
+`;
+
+
+function handleContentClick(e) {
+    //content section clicked, prevent modal from closing.
+    e.stopPropagation();
+};
+
+function handleKeyUp(e, props: ModalProps) {
+    //escape pressed, close modal.
+    if (e.keyCode === 27) props.handleClose();
+}
+
+function closeIconKeyUp(e, cb) {
+    //close (X) icon key up, if enter/space, call handleClose.
+    if (e.keyCode === 13 || e.keyCode === 32) {
+        cb();
+    };
+};
+
+function createHeader(props: ModalProps) {
+    return (
+        <HeaderWrapper>
+            {props.title ? 
+                <StyledTitle
+                    tabIndex={1}
+                    className={props.titleClass}
+                    style={props.titleStyle}
+                >
+                    {props.title}
+                </StyledTitle>
+            : undefined}
+            <CloseIcon
+                onKeyUp={(e) => closeIconKeyUp(e, props.handleClose)}
+                tabIndex={1}
+                onClick={props.handleClose}
+            />
+        </HeaderWrapper>
+    );
+};
+
+export class Modal extends React.Component<ModalProps> {
+    
+    private modal;
+
+    public componentDidMount(): void {
+        //create div and apply directly to body of document.
+        this.modal = document.createElement('div');
+        document.body.appendChild(this.modal);
+    };
+    
+    public componentWillUnmount(): void {
+        //remove modal from body of document.
+        ReactDOM.unmountComponentAtNode(this.modal);
```

Comment:
I have some ideas for a potentially cleaner solution for modals that doesn't require rendering outside of the main React pipeline.

Reply:
:+1: Nice solution, man. We still need to figure out how we get the data from the modal back to where the modal was launched from (without Redux).